src/h5web/dataset-visualizer/vis/MatrixVis.tsx

```diff
@@ -13,34 +14,162 @@ interface Props {
   const { dims, data } = props;
+  const gridDimensions = {
+    rowHeight: 32,
+    rowCount: dims[0] + 1,
+    columnWidth: 116,
+    columnCount: (dims.length === 2 ? dims[1] : 1) + 1,
+  };
+  const GridSettings = createContext(gridDimensions);
+
+  const CellRenderer: CellRendererFunc = (
+    cellRendererProps: GridChildComponentProps
+  ) => {
+    const { data: itemData, rowIndex, columnIndex, style } = cellRendererProps;
+    const { NormalCellFunc } = itemData;
+    // Disable rendering of index columns (done by the innerElementType)
+    if (rowIndex * columnIndex === 0) {
+      return <></>;
+    }
+    return (
+      <NormalCellFunc
+        rowIndex={rowIndex - 1}
+        columnIndex={columnIndex - 1}
+        style={style}
+      />
+    );
+  };
 
   const accessor: AccessorFunc =
     dims.length === 1 ? row => data[row] : (row, col) => data[row][col];
 
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  function ValueCell(cellProps: CellProps): JSX.Element {
+    const { rowIndex: row, columnIndex: col, style } = cellProps;
+    return (
+      <div
+        className={styles.cell}
+        style={style}
+        data-bg={(col + row) % 2 === 1 ? '' : undefined}
+      >
+        {numeral(accessor(row, col)).format('0.000e+0')}
+      </div>
+    );
+  }
+
+  function IndexCell(indexCellProps: IndexCellProps): JSX.Element {
+    const { index, style } = indexCellProps;
+    return (
+      <div className={styles.indexCell} style={style}>
+        {index >= 0 && index}
+      </div>
+    );
+  }
+
+  function IndexColumn(): JSX.Element {
+    const { columnWidth, rowHeight, rowCount } = useContext(GridSettings);
+    return (
+      <div
+        className={`${styles.sticky} ${styles.indexColumn}`}
+        style={{ width: columnWidth }}
+      >
+        {range(rowCount - 1).map(row => {
+          return (
+            <IndexCell
+              index={row}
+              style={{
+                width: columnWidth,
+                height: rowHeight,
+              }}
+            />
+          );
+        })}
+      </div>
+    );
+  }
+
+  function IndexRow(): JSX.Element {
+    const { columnWidth, rowHeight, columnCount } = useContext(GridSettings);
+    return (
+      <div
+        className={`${styles.sticky} ${styles.indexRow}`}
+        style={{ height: rowHeight }}
+      >
+        {range(columnCount).map(col => {
+          return (
+            <IndexCell
+              // Starts a -1 as the first column is the index column
+              index={col - 1}
+              style={{
+                width: columnWidth,
+                height: rowHeight,
+              }}
+            />
+          );
+        })}
+      </div>
+    );
+  }
+
+  const innerElementType = forwardRef(({ children, ...rest }, ref) => {
+    return (
+      <div ref={ref as Ref<HTMLDivElement>} {...rest}>
+        <IndexRow />
+        <IndexColumn />
+        {children}
+      </div>
+    );
+  });
```

Comment:
I wonder if all functions that are only relevant to `IndexedGrid` should be moved inside this function.

Reply:
See https://github.com/silx-kit/h5web/pull/28#discussion_r397061137